analyzers/tests/SonarAnalyzer.Test/TestCases/AspNet/ControllerReuseClient.CSharp12.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+[ApiController]
+[Route("SomeRoute")]
+public class C(HttpClient client) : ControllerBase;
+
+[ApiController]
+[Route("SomeRoute")]
+public class D(HttpClient client) : C(new HttpClient()) // Compliant
```

Comment:
It's in the constructor - it can be re-used in the context of one request.
This is the reason why we consider compliant also assignment to properties and fields.